kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -616,6 +621,28 @@ def _submit_container_state_metric(self, metric_name, state_name, c_state, state
             gauge_name = '{}.containers.{}.{}'.format(self.NAMESPACE, metric_name, state_name)
             self.gauge(gauge_name, 1, tags + reason_tags)
 
+    def _update_kubelet_url_and_bearer_token(self, instance, endpoint): 
+        if 'cadvisor_metrics_endpoint' in instance:
+            self.cadvisor_scraper_config['prometheus_url'] = instance.get('cadvisor_metrics_endpoint', urljoin(endpoint, CADVISOR_METRICS_PATH))
+        else:
+            self.cadvisor_scraper_config['prometheus_url'] = instance.get('metrics_endpoint', urljoin(endpoint, CADVISOR_METRICS_PATH))
+        self.cadvisor_scraper_config["bearer_token_auth"] = instance["bearer_token_auth"]
+        self.set_bearer_token(self.cadvisor_scraper_config)
+
+        self.kubelet_scraper_config['prometheus_url'] = instance.get('kubelet_metrics_endpoint', urljoin(endpoint, KUBELET_METRICS_PATH))
+        self.kubelet_scraper_config["bearer_token_auth"] = instance["bearer_token_auth"]
```

Comment:
Is the explicit setting of `bearer_token_auth` necessary in this function? It seems like that is happening in `set_bearer_token`

Reply:
The reason I did it this way is because (as far as I can tell) the Kubelet check runs three openmetrics checks, one each for a different set of metrics. The `instance` value here is the original `bearer_token_auth` value; and we have to set the specific config of all three checks (`kubelet_scraper_config`, `cadvisor_scraper_config`, and `probes_scraper_config`) to the original `bearer_token_auth` value (which is in `instance`) before they can be converted based on the new url.

Reply:
I see. I think we should try to limit the overall number of config values we need to manually change ourselves outside of `set_bearer_token`, as otherwise it makes calling this function more difficult.

I added another comment [here](https://github.com/DataDog/integrations-core/pull/14546#discussion_r1200744108) which might have a workaround for this